src/System.Private.Xml.Linq/tests/TreeManipulation/SaveWithFileName.cs

```diff
@@ -0,0 +1,287 @@
+            Assert.Throws<ArgumentNullException>(() => new XElement("e").Save((string)null, SaveOptions.None));
+        }
+
+        [Fact]
+        public void XStreamingElementSave_NullParameter()
+        {
+            Assert.Throws<ArgumentNullException>(() => new XStreamingElement("e").Save((string)null));
+            Assert.Throws<ArgumentNullException>(() => new XStreamingElement("e").Save((string)null, SaveOptions.DisableFormatting));
+            Assert.Throws<ArgumentNullException>(() => new XStreamingElement("e").Save((string)null, SaveOptions.None));
+        }
+
+        [Fact]
+        public void XStreamingElementSave()
+        {
+            string markup = "<e a=\"value\"> <!--comment--> <e2> <![CDATA[cdata]]> </e2> <?pi target?> </e>";
+            try
+            {
+                XElement e = XElement.Parse(markup, LoadOptions.PreserveWhitespace);
+                XStreamingElement e2 = new XStreamingElement(e.Name, e.Attributes(), e.Nodes());
+                e2.Save(_fileName);
+            }
+            finally
+            {
+                Assert.True(System.IO.File.Exists(_fileName));
+                Assert.Equal("<?xml version=\"1.0\" encoding=\"utf-8\"?>\r\n" + markup, File.ReadAllText(_fileName));
+                System.IO.File.Delete(_fileName);
+            }
+        }
+
+        [Fact]
+        public void XStreamingElementSave_SaveOptions()
+        {
+            string markup = "<e a=\"value\"> <!--comment--> <e2> <![CDATA[cdata]]> </e2> <?pi target?> </e>";
+            try
+            {
+                XElement e = XElement.Parse(markup, LoadOptions.PreserveWhitespace);
+                XStreamingElement e2 = new XStreamingElement(e.Name, e.Attributes(), e.Nodes());
+                e2.Save(_fileName, SaveOptions.DisableFormatting);
+            }
+            finally
+            {
+                Assert.True(System.IO.File.Exists(_fileName));
+                Assert.Equal("<?xml version=\"1.0\" encoding=\"utf-8\"?>" + markup, File.ReadAllText(_fileName));
+                System.IO.File.Delete(_fileName);
+            }
+        }
+
+        //
+        // helpers
+        //
+        private static string SerializeXDocumentSaveToFile(XNode node)
+        {
+            string filename = System.IO.Path.GetTempFileName();
+            string result;
+            try
+            {
+                ((XDocument)node).Save(filename);
+                using (System.IO.TextReader tr = new System.IO.StreamReader(filename))
+                {
+                    result = StripOfXmlDeclaration(tr.ReadToEnd());
+                }
+            }
+            finally
+            {
+                Assert.True(System.IO.File.Exists(filename));
+                System.IO.File.Delete(filename);
+            }
+            return result;
+        }
+
+        private static string SerializeXElementSaveToFile(XNode node)
+        {
+            string filename = System.IO.Path.GetTempFileName();
```

Comment:
> System.IO.Path.GetTempFileName(); [](start = 29, length = 34)

this creates a file IIRC so strictly it needs to be inside the try so it gets deleted